core/src/main/scala/sttp/tapir/internal/package.scala

```diff
@@ -90,12 +91,16 @@ package object internal {
         case s: EndpointOutput.OneOf[_] =>
           Right(
             ListMap(
-              s.mappings
+              (s.mappings
                 .map(c => (c.output.asBasicOutputsOrMap, c.statusCode))
-                .map {
-                  case (Left(basicOutputs), statusCode) => statusCode -> basicOutputs
-                  case (Right(_), _)                    => throwMultipleOneOfMappings
-                }: _*
+                .foldLeft(Map.empty[Option[StatusCode], BasicOutputs]) { (acc, e) =>
```

Comment:
this implementation "flattens" all of the repeated alternatives into a single list. So you might end up with a single status code mapped e.g. to multiple `Body` outputs.

While this can work fine for documentation (mostly by accident), it would cause weird bugs if used e.g. in servers or clients. I think the correct implementation here would be to return a `Map[Option[StatusCode], List[BasicOutputs]]` - making it explicit that there might be multiple variants of mappings for a single status code. 

Or using some implementation of a multi-map instead of a normal map.

Then, the users of this method would be able to decide, how to treat cases when there are multiple mappings for a single status code (take the first / throw an error etc.)

Reply:
Sorry to reply to a 9 months old comment but I'm facing the same original issue and I would have solved the problem in a similar way.

I actually don't get your point: `BasicOutputs` is a `Vector[EndpointOutput.Basic[_]]` so it can already be used to map a single status code to multiple `Body` outputs. Or did I miss something?